internal/collections/syncmap.go

```diff
@@ -14,7 +14,11 @@ func (s *SyncMap[K, V]) Load(key K) (value V, ok bool) {
 	if !ok {
 		return
 	}
-	return val.(V), true
+	if value, ok := val.(V); ok {
+		return value, true
+	}
+	var zero V
+	return zero, true
```

Comment:
This type assertion failing should be impossible; the only way to store a value into the map is to have passed in a `V`, which is enforced by the generics, no?

The only way that I can see this failing is if we are accidentally converting one SyncMap to another, which, I accidentally think is possible because I forgot to  mention `K` and `V` in the struct...

Reply:
Are you sure the issue isn't somewhere else where we shouldn't be boxing a nil pointer into an interface? I'm just not seeing how it's possible for this to crash at runtime at all

Reply:
We are calling file exists on package .json which caches nil as stat for that file .. then to watch that location we get stats and record its mTime which gets you the cached stat that’s nil

Reply:
Sorry, I'm not understanding... The only way to get a value into `m` is via `Store` or `LoadOrStore`, both of which only accept a `V`. How can the underlying value possibly be anything other than `V`?